src/main/java/org/elasticsearch/common/jna/Natives.java

```diff
@@ -62,6 +65,42 @@ public static void tryMlockall() {
         }
     }
 
+    public static void tryVirtualLock()
+    {
+        Kernel32Library kernel = Kernel32Library.getInstance();
+        Pointer process = null;
+        try {
+            process = kernel.GetCurrentProcess();
+            // By default, Windows limits the number of pages that can be locked.
+            // Thus, we need to first increase the working set size of the JVM by
+            // the amount of memory we wish to lock (Xmx), plus a small overhead (1MB).
+            SizeT size = new SizeT(JvmInfo.jvmInfo().getMem().getHeapInit().getBytes() + (1024 * 1024));
```

Comment:
If they are different then mlockall will not really work on unix either. That is because it may map additional stuff later!